WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore.swift

```diff
@@ -461,6 +481,54 @@ public final class ConversationLocalStore: ConversationLocalStoreProtocol {
             )
         }
     }
+    
+    public func updateTypingUsers(
+        conversationID: NSManagedObjectID,
+        usersID: Set<NSManagedObjectID>
+    ) async {
+        
+        // Switching to main context
+        let uiContext = await context.perform { [context] in
```

Comment:
it's not completely clear to me why this needs to be done on viewContext, any impact if we change the context?

Reply:
typing users (ZMUser) managed object are directly consumed by the UI component. Legacy code is not very straightforward, this could be simplified but as mentioned above this involves changing a little bit the app layer logic (specifically the `ConversationInputBarViewController`)
 
view.typingUsers = conversation.typingUsers 
> gets the typing users object nested in the main context user info dictionary ->  in that object it gets the users object ids  and map them as ZMUser

I'm not sure why we need all that stuff, I could start a refactoring of this legacy code  but I'm not 100% sure we won't break anything by doing so and also I don't think it's in the scope of that PR

What do you think ? also mentioning @johnxnguyen  for the discussion